gpflow/likelihoods/base.py

```diff
@@ -294,6 +295,86 @@ def variational_expectations(self, Fmu, Fvar, Y):
     def _variational_expectations(self, Fmu, Fvar, Y):
         raise NotImplementedError
 
+    # @abc.abstractmethod
+    def conditional_sample(self, Fsample) -> tf.Tensor:
+        """ Sample from the likelihood """
+
+
+class ConditionedLikelihood:
+    """
+    This class represents the likelihood of the output conditioned on the latent
+    process(es) function values.
+    """
+
+    def __init__(self, likelihood: Likelihood, Fmean: tf.Tensor, Fvar: tf.Tensor) -> None:
+        """
+        :param likelihood: model likelihood
+        :param Fmean: latent process mean values obtained at some points X.
+        :param Fvar: latent process variance values ontained at some points.
+        """
+        self.likelihood = likelihood
+        self.f_mean = Fmean
+        self.f_var = Fvar
+
+    def _get_f_samples(self, num_samples: int) -> tf.Tensor:
+        """ Convenience method to return samples from the latent process """
+        return sample_mvn(self.f_mean, self.f_var, full_cov=False, num_samples=num_samples)
+
+    def parameter_samples(self, num_samples: int = 1000) -> Tuple[tf.Tensor, ...]:
+        """ Get samples from the distribution of likelihood parameters """
+        f_sample = self._get_f_samples(num_samples)
+        return self.likelihood.conditional_parameters(f_sample)
+
+    def mean_and_var(self):
+        """ Return the (conditional) output mean and variance """
+        return self.likelihood.predict_mean_and_var(self.f_mean, self.f_var)
+
+    def log_density(self, Y: tf.Tensor) -> tf.Tensor:
+        """
+        Return the (conditional) log density of the output values.
```

Comment:
I think it would be helpful to add the equations to the docstrings to make clear what exactly is being computed here. I'm not sure "(conditional)" is correct in this case.